### DISCUSSION TITLE
How to set SO_SNFBUF and SO_SNDRCV for Datagram socket

### DISCUSSION THREAD

**Discussion by Lulu300** [agronholm/anyio#449] [Q&A] [answered]
Hi all,

I was wondering how to set SO_SNFBUF and SO_SNDRCV option for an udp socket ?
I looked through the documentation but didn't see an option for this.

Thanks in advance for anybody that can help me with this

**Comment by agronholm** [comment, marked as answer]
You can get the raw socket and then use `setsockopt()` on it:
```python
import socket
from anyio import SocketAttribute

raw_socket = udpsocket.extra(SocketAttribute.raw_socket)
raw_socket.setsockopt(socket.SOL_SOCKET, socket.SO_SNDBUF, 65536)
```

**Reply by Lulu300** [reply]
Thank you, it works perfectly !
